src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -371,19 +372,21 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema)(im
     } else {
 
       val futurePerRequestMessage = for {
-        arrowResponse <- callArrow(pfbUrl.toString)
+        arrowResponse <- callArrow
         rawlsJsonEntity <- arrowResponse.status match {
           case OK => Future.successful(arrowResponse.entity)
           case _ => Future.failed(new FireCloudException(arrowResponse.status.value + ": " + arrowResponse.entity.asString))
         }
         rawlsResponse <- callRawls(rawlsJsonEntity)
         result <- rawlsResponse.status match {
           case NoContent => Future.successful(RequestComplete(NoContent))
+          case NotFound => Future.failed(new FireCloudExceptionWithErrorReport(FCErrorReport(rawlsResponse)))
           case _ => Future.failed(new FireCloudException(rawlsResponse.status.value + ": " + rawlsResponse.entity.asString))
```

Comment:
I still don't understand why the `NotFound` case couldn't be handled by the catchall case. Writing it somewhat differently:
```
case NoContent => Future.successful(RequestComplete(NoContent))
case errorCode:StatusCode =>
            val errorReport = FCErrorReport.tryUnmarshal(rawlsResponse) match {
              case Success(rpt) => rpt
              case Failure(_) => ErrorReport(errorCode, "Unexpected failure: " + rawlsResponse.entity.asString)
            }
            Future.failed(new FireCloudExceptionWithErrorReport(errorReport))
```
`FCErrorReport.tryUnmarshal` doesn't look for a successful status code. And in your current PR, `FCErrorReport(rawlsResponse)` could fail if the response body is invalid.

https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/core/GetEntitiesWithType.scala#L72 is an example usage.

Here again - tests could help validate what the code does, and I'm okay with the current PR if it needs to go in as-is for deadline reasons.

Reply:
`tryUnmarshal` calls spray's `unmarshal` function which looks like:
```
  def unmarshal[T: FromResponseUnmarshaller]: HttpResponse ⇒ T =
    response ⇒
      if (response.status.isSuccess)
        response.as[T] match {
          case Right(value) ⇒ value
          case Left(error: MalformedContent) ⇒
            throw new PipelineException(error.errorMessage, error.cause.orNull)
          case Left(error) ⇒ throw new PipelineException(error.toString)
        }
      else throw new UnsuccessfulResponseException(response)
```
Interestingly, after writing some tests and plugging in the code you suggested, most of the tests passed. The one that failed was one where I stubbed Rawls to return a 400 response which I want to convert to 500 (it's a bad request from orch, not from the caller of `importPFB`). For the ones that succeeded, I set some breakpoints and attached a debugger and, as I expected, they always match the `Failure(_)` case. So, the code using `tryUnmarshal` has the right behavior, but the code feels misleading as to why. I believe, other than for 20x status codes, `tryUnmarshal` is guaranteed to throw an exception.

I'm happy to discuss further. For the moment, I'm moving forward with the code as it is (with a minor change to also handle 401 and 403 responses similar to 404s).